doc/userguide/devguide/internals/engines/stream/applayertriggerrawreassembly.rst

```diff
@@ -0,0 +1,67 @@
+Triggering Raw Stream Reassembly from Application Layer
+-------------------------------------------------------
+
+It is possible to trigger the reassembly of the raw stream from the application
+layer protocol parsers, by calling the function
+``AppLayerParserTriggerRawStreamReassembly``.
+This call is made for the immediate reassembly of raw stream to make more data
+available for detection, e.g. upon completion of an HTTP request because of the
+following reasons:
+
+1. The reassembly ideally only happens after a certain chunk of data has been
+   seen by the Stream Engine. This chunk size is configurable in
+   ``suricata.yaml`` (see: ``stream.reassembly.toserver-chunk-size``).
+2. The Stream Engine would have seen more data than the Detection Engine at any
+   point in time.
+
+The Stream Engine provides the following parameters for the stream to be
+tracked by the different subsystems of Suricata:
+
+* app layer progress
+* raw reassembly progress
+* streaming logger progress
+
+These parameters are used to track the progress of the stream in the respective
+subsystems. This serves as a communication between the different subsystems
+about how far they have already looked into the stream. They are all relative
+to the base offset of the streaming buffer region as seen in the file
+``stream-tcp-private.h``.
+
+.. code-block:: c
+
+    #define STREAM_APP_PROGRESS(stream) (STREAM_BASE_OFFSET((stream)) + (stream)->app_progress_rel)
+    #define STREAM_RAW_PROGRESS(stream) (STREAM_BASE_OFFSET((stream)) + (stream)->raw_progress_rel)
+    #define STREAM_LOG_PROGRESS(stream) (STREAM_BASE_OFFSET((stream)) + (stream)->log_progress_rel)
+
+For a given streaming buffer region as illustrated below:
+
+.. image:: stream-engine/trigger_reassembly_before.png
+
+If a call to ``AppLayerParserTriggerRawStreamReassembly`` was made, things
+would change as represented in the next image, allowing the Detection Engine to
+see a bigger part of the stream until the last ACK as seen by the Stream
```

Comment:
this gets things backwards. The goal of trigger reassembly is to trigger inspection sooner, so for less data. This is because a HTTP request (for example) maybe be smaller than the chunk size. So it needs to be inspected sooner, so be inspected together with that requests tx keywords.

Reply:
hmm that's what I intended to show. Could you please tell what in the image indicates that it is at/beyond the chunk size?